docs/connections.rst

```diff
@@ -0,0 +1,335 @@
+that one should keep in mind when using decoded connections.
+
+- Decoders will be automatically solved for in the build process.
+
+  Solving for decoders makes up the majority of build time,
+  so if building your networks takes a long time,
+  look at your decoded connections and
+  try lowering the number of neurons
+  or using different `.Solver` types.
+
+- The function passed to the connection
+  is used to determine decoders.
+  It will never be run during the simulation.
+
+  When you define a function in a node,
+  it will be execute on every simulation timestep.
+  That may lead you to think that the function
+  passed to a connection is executed on every timestep,
+  but that is *not* the case for decoded connections.
+  The function passed to the connection will be executed
+  in the decoder solving process determine an error
+  to minimize, but never during the simulation.
+
+- The characteristics of the ``pre`` ensemble
+  are critically important to performance.
+
+  If you determine that your decoded connection
+  is not approximating the desired function well,
+  examine the ``pre`` ensemble.
+  The decoded value depends on the activity
+  of the ``pre`` ensemble;
+  does it represent its input reliably?
+  If not, then a function of that input
+  cannot be well approximated.
+  If you think that your function may be incorrect,
+  switch the ``pre`` ensemble to use
+  the `.Direct` neuron type,
+  which does not use decoders.
+  If that function looks correct,
+  move on to a simpler neuron type
+  like `.RectifiedLinear` until you
+  can determine why your function is not
+  being approximated well.
+
+Direct connections
+==================
+
+Any connection that is not a decoded connection
+is a direct connection.
+
+For simplicity and consistency,
+Nengo exposes the same interface
+for decoded and direct connections.
+In all cases, data from the ``pre`` object
+is sent to the ``post`` object,
+with an optional ``synapse`` filter.
+In decoded connections,
+weights are automatically determined
+through decoder solving.
+In direct connections,
+weights can be manually specified
+through the ``transform`` argument. [1]_
+
+The most common example of a direct connection
+is a neuron-to-neuron connection.
+These connections are the types of connections
+you see in most neural simulators,
+and can be used to reproduce networks
+written in other simulators like
+`Brian <http://briansimulator.org/>`_::
+
+  with nengo.Network() as net:
+      ens1 = nengo.Ensemble(10, dimensions=1)
+      ens2 = nengo.Ensemble(20, dimensions=2)
+
+      # Neuron to neuron
+      weights = np.random.normal(size=(ens2.n_neurons, ens1.n_neurons))
+      nengo.Connection(ens1.neurons, ens2.neurons, transform=weights)
+
+Note that, unlike with decoded connections,
+it does not matter that the dimensionality of ``ens1``
+does not match the dimensionality of ``ens2``.
+It only matters that the ``transform``
+is of the correct shape,
```

Comment:
But isn't that true for decoded connections too? If the transform is of the correct shape, the dimensionalities of the ensembles do not matter.